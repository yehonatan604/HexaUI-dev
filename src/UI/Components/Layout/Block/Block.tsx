import useTheme from "../../../../Core/Context/ThemeContext/hooks/useTheme";
import { TBlock } from "./types/TBlock";

const Block = (props: TBlock) => {
  const { options, inline, children, className, innerClassName, ...componentProps } =
    props;
  const { mode } = useTheme();

  // Options
  const bgColor = options?.bgVariant || "transparent";
  let textColor = mode === "dark" ? "white-d" : "black-l";
  if (options?.textVariant) textColor = options.textVariant as string;
  const borderColor = options?.border?.variant || "transparent";
  const borderWidth = options?.border?.thickness || "border";
  const borderRadius = options?.border?.radius || "rounded";
  const block = inline ? "inline-block" : "block";
  const padding = options?.padding || "4";
  const innerPadding = options?.innerPadding || "4";

  console.log("Block -> options", options);
  console.log(textColor);

  // JSXf
  return (
    <div
      className={`${block} p-${padding} bg-${bgColor} text-${textColor} border-${borderColor} ${borderWidth} ${borderRadius} ${className}`}
      {...componentProps}
    >
      <div
        className={`${inline ? "flex items-center" : block} text-${textColor} p-${innerPadding} space-y-${!inline && innerPadding} gap-${inline && innerPadding} ${innerClassName}`}
      >
        {children}
      </div>
    </div>
  );
};

export default Block;
