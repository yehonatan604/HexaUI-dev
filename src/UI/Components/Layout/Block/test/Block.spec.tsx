import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Block from "../Block";
import ThemeProvider from "../../../../../Core/Context/ThemeContext/providers/Theme.provider";

// Sample props to test different scenarios
const sampleProps = {
  options: {
    bgVariant: "blue-500",
    textVariant: "gray-900",
    border: {
      variant: "blue-300",
      thickness: "border-2",
      radius: "rounded-lg",
    },
    padding: "6",
    innerPadding: "2",
  },
  inline: false,
  className: "custom-class",
  innerClassName: "inner-custom-class",
};

describe("Block Component", () => {
  it("should render with default props", () => {
    const { container } = render(
      <ThemeProvider>
        <Block>Default Block</Block>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with custom options", () => {
    const { container } = render(
      <ThemeProvider>
        <Block {...sampleProps}>Custom Block</Block>
      </ThemeProvider>
    );

    expect(container.querySelector(".bg-blue-500")).toBeDefined();
    expect(container.querySelector(".text-gray-900")).toBeDefined();
    expect(container.querySelector(".border-blue-300")).toBeDefined();
    expect(container.querySelector(".border-2")).toBeDefined();
    expect(container.querySelector(".rounded-lg")).toBeDefined();
    expect(container.querySelector(".p-6")).toBeDefined();
    expect(container.querySelector(".inner-custom-class")).toBeDefined();
  });

  it("should use theme context to determine text color", () => {
    const { container } = render(
      <ThemeProvider>
        <Block>Dark Mode Block</Block>
      </ThemeProvider>
    );

    expect(container.querySelector(".text-white-d")).toBeDefined();
  });

  it("should apply inline styles correctly", () => {
    const { container } = render(
      <ThemeProvider>
        <Block inline>Inline Block</Block>
      </ThemeProvider>
    );

    expect(container.querySelector(".inline-block")).toBeDefined();
  });
});
