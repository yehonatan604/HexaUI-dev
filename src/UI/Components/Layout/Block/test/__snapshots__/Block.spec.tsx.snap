// Vitest Snapshot v1

exports[`Block Component > should render with default props 1`] = `
<div>
  <div
    class="block p-4 bg-transparent text-black-l border-transparent border rounded undefined"
  >
    <div
      class="block text-black-l p-4 space-y-4 gap-undefined undefined"
    >
      Default Block
    </div>
  </div>
</div>
`;
